components/clp-package-utils/clp_package_utils/general.py

```
@@ -166,6 +170,41 @@ def get_clp_home():
+    :param data: Request body as bytes.
+    :param headers: Request headers.
+    :param method: HTTP method.
+    :param max_attempts: Number of attempts before giving up.
+    :param retry_delay: Seconds to wait between retries.
+    :param timeout: Seconds to wait for a response per attempt.
+    :return: The HTTP response.
+    :raise urllib.error.URLError: If all attempts fail due to a network-level error.
+    :raise urllib.error.HTTPError: If all attempts fail with an HTTP error status.
+    :raise OSError: If all attempts fail due to a socket-level error.
+    """
+    req = urllib.request.Request(url, data=data, headers=headers or {}, method=method)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)